mypy_django_plugin/transformers/managers.py

```diff
@@ -175,6 +176,67 @@ def resolve_manager_method(ctx: AttributeContext) -> MypyType:
         return AnyType(TypeOfAny.from_error)
 
 
+def merge_queryset_into_manager(
+    api: SemanticAnalyzer,
+    manager_info: TypeInfo,
+    queryset_info: TypeInfo,
+    manager_base: TypeInfo,
+    class_name: str,
+) -> Optional[TypeInfo]:
+    """
+    Merges a queryset definition with a manager definition.
+    As a reference one can look at the 3-arg call definition for ``builtins.type`` to
+    get an idea of what kind of class we're trying to express a type for.
+    """
+    # Stash the queryset fullname so that our 'resolve_manager_method' attribute hook
+    # can fetch the method from that QuerySet class
+    manager_info.metadata["django"] = {"from_queryset_manager": queryset_info.fullname}
+
+    manager_base.metadata.setdefault("from_queryset_managers", {})
+    # The `__module__` value of the manager type created by Django's `.from_queryset`
+    # is `django.db.models.manager`. But `basic_new_typeinfo` defaults to what is
+    # currently being processed, so we'll map that together through metadata.
+    manager_base.metadata["from_queryset_managers"][f"django.db.models.manager.{class_name}"] = manager_info.fullname
+
+    # So that the plugin will reparameterize the manager when it is constructed inside of a Model definition
+    helpers.add_new_manager_base(api, manager_info.fullname)
+
+    self_type = fill_typevars(manager_info)
+    assert isinstance(self_type, Instance)
+
+    # We collect and mark up all methods before django.db.models.query.QuerySet as class members
+    for class_mro_info in queryset_info.mro:
+        if class_mro_info.fullname == fullnames.QUERYSET_CLASS_FULLNAME:
+            break
+        for name, sym in class_mro_info.names.items():
+            if not isinstance(sym.node, (FuncDef, Decorator)):
+                continue
+            # Insert the queryset method name as a class member. Note that the type of
+            # the method is set as Any. Figuring out the type is the job of the
+            # 'resolve_manager_method' attribute hook, which comes later.
+            #
+            # class BaseManagerFromMyQuerySet(BaseManager):
+            #    queryset_method: Any = ...
+            #
+            helpers.add_new_sym_for_info(
+                manager_info,
+                name=name,
+                sym_type=AnyType(TypeOfAny.special_form),
```

Comment:
```suggestion
                sym_type=AnyType(TypeOfAny.special_form),
```
```suggestion
                sym_type=AnyType(TypeOfAny.implementation_detail),
```

Reply:
We have a test case `from_queryset_can_resolve_explicit_any_methods` doing that already. I think that should suffice

https://github.com/typeddjango/django-stubs/blob/2a6f4647f083b2a9cd728b7e5698dc76c59b5449/tests/typecheck/managers/querysets/test_from_queryset.yml#L333-L360